io/src/main/scala/fs2/io/io.scala

```diff
@@ -124,4 +124,30 @@ package object io {
     */
   def toInputStream[F[_]](implicit F: Effect[F], ec: ExecutionContext): Pipe[F,Byte,InputStream] =
     JavaInputOutputStream.toInputStream
+
+  /**
+    * Reads all the values of the iterator into a Stream.
+    */
+  def readIterator[F[_], A](iterator: Iterator[A])(implicit F: Effect[F]): Stream[F, A] = {
+    def getNext(i: Iterator[A]): F[Option[(A, Iterator[A])]] =
+      F.delay(i.hasNext).flatMap(hasNext =>
+        if (hasNext) F.delay(i.next()).map(a => (a, i).some)
+        else F.pure(None)
+      )
+
+    Stream.unfoldEval(iterator)(getNext)
+  }
+
+  /**
+    * Brackets Resource Acquisition for an Iterator
+    */
+  def readBracketedIterator[F[_]: Effect, R, O](acquire: F[R])
```

Comment:
I'm not convinced we need this method. If we have `Stream.fromIterator`, I think putting that together with `Stream.bracket` is simple enough.

Reply:
I'm also a bit unconvinced we need this, especially because the user still has to do some massaging to get an `R => F[Iterator[O]]` (the wrapping in `F`), at which point one could use `bracket` directly